src/java_bytecode/java_types.h

```diff
@@ -609,12 +609,18 @@ class java_generic_symbol_typet : public symbol_typet
   /// in the vector of generic types.
   /// \param type The type we are looking for.
   /// \return The index of the type in the vector of generic types.
-  optionalt<size_t> generic_type_index(const reference_typet &type) const
+  optionalt<size_t>
+  generic_type_index(const java_generic_parametert &type) const
```

Comment:
If you were to write a trivial unit test for this function that would be :ok_hand: to easily see why it was broken before but not now.